### DISCUSSION TITLE
Extend getPropertyValue return value by itemCount

### DISCUSSION THREAD

**Discussion by edarroudi** [microsoft-search/pnp-modern-search#1577] [Q&A] [answered]
Hi,

can we extend the returned value with the count of items found in the result source? I want to use that information in other Webparts and i think i'm not able to retrieve that info from the handlebar Context.

`
public getPropertyValue(propertyId: string) {

        switch (propertyId) {
            case ComponentType.SearchResults:

                // Pass the Handlebars context to consumers, so they can register custom helpers for their own services 
                this._currentDataResultsSourceData.handlebarsContext = this.templateService.Handlebars;

                return this._currentDataResultsSourceData;

            case DynamicDataProperties.AvailableFieldValuesFromResults:

                // Dynamic data values should be flatten https://docs.microsoft.com/en-us/sharepoint/dev/spfx/dynamic-data
                let fields = {};
                this._currentDataResultsSourceData.availableFieldsFromResults.forEach((field: string) => {
        
                    // Aggregate all values for this specific field across all items
                    // Ex:
                    // "FileType":['docx','pdf']
                    let fieldValue = undefined;
        
                    if (this._currentDataResultsSourceData.selectedItems.length > 0) {
                        fieldValue = ObjectHelper.byPath(this._currentDataResultsSourceData.selectedItems, field);
                    }
                    
                    fields[field] = fieldValue ? fieldValue.split(',') : undefined;
                });
        
                return fields;
        }

        throw new Error('Bad property id');
    }
`

**Comment by FranckyC** [comment, marked as answer]
Hi @edarroudi , yes sure, just add a new property in the `IDataResultSourceData` interface and set it in the `getPropertyValue` method.

**Reply by edarroudi** [reply]
Hey @FranckyC , cool. I did it quickly and it worked like a charm. If you need more tests / information i'll add it.
https://github.com/microsoft-search/pnp-modern-search/pull/1578

**Reply by FranckyC** [reply]
@edarroudi can you mark this as answer so we can close it. Thanks!
